symfit/core/fit_results.py

```diff
@@ -146,5 +187,86 @@ def _array_safe_dict_eq(one_dict, other_dict):
                 return False
         else: return True
 
-    def __eq__(self, other):
-        return FitResults._array_safe_dict_eq(self.__dict__, other.__dict__)
\ No newline at end of file
+    def __getstate__(self):
+        state = self.__dict__.copy()
+
+        if hasattr(state['minimizer'], 'minimizers'):  # ChainedMinimizer
+            minimizer_cls = [type(minimizer)
+                             for minimizer in state['minimizer'].minimizers]
+        else:
+            minimizer_cls = type(state['minimizer'])
+        state['minimizer'] = (minimizer_cls,
+                              state['minimizer'].objective,
+                              state['minimizer'].parameters)
+        return state
+
+    def __setstate__(self, state):
+        min_class, objective, parameters = state['minimizer']
+        try:
+            # If min_class is iterable, initiate a ChainedMinimizer.
+            minimizers = [cls(objective, parameters) for cls in min_class]
+        except TypeError:
+            state['minimizer'] = min_class(objective, parameters)
+        else:
+            # Has to be imported here to prevent circular imports
+            from symfit.core.minimizers import ChainedMinimizer
+            state['minimizer'] = ChainedMinimizer(objective, parameters,
```

Comment:
use `min_class` instead?

Reply:
`min_class` is an iterable of minimizer types in this case, corresponding to the `.minimizers` of the original ChainedMinimizer.

Reply:
turns out that ChainedMinimizer already works directly, so this can be simplified. Stay tuned.